pkg/controllers/allocation/bind.go

```diff
@@ -59,10 +65,41 @@ func (b *Binder) Bind(ctx context.Context, node *v1.Node, pods []*v1.Pod) error
 		if !errors.IsAlreadyExists(err) {
 			return fmt.Errorf("creating node %s, %w", node.Name, err)
 		}
+		// If the node object already exists, make sure finalizer and taint are in place.
+		if err := b.KubeClient.Patch(ctx, node, client.StrategicMergeFrom(stored)); err != nil {
+			return fmt.Errorf("patching node %s, %w", node.Name, err)
+		}
 	}
-
-	// 4. Bind pods
 	errs := make([]error, len(pods))
+	// 4. Wait for node readiness.
```

Comment:
We can't wait here. This will block the entire provisioning process on nodes coming online, which will massively reduce provisioning throughput. If we move forward with this path, we will need some async mechanism for this.

Reply:
I pushed a commit which uses asynchronous bind, but only if it is required.